[GStreamer] Fix content type frame rate limit

eocanha opened the pull request:
#### 2505dd003363bb15630b09de3570e750aeb6ebb4
<pre>
[GStreamer] Fix content type frame rate limit
<a href="https://bugs.webkit.org/show_bug.cgi?id=296572">https://bugs.webkit.org/show_bug.cgi?id=296572</a>

Reviewed by Philippe Normand.

Decoder limits are set by a build configuration in the form of
widthxheight@framerate. However, framerate limit is only considered when
the width and height are the ones in the build config. If the actual
requested resolution is different, no framerate limit is ever
considered.

See: <a href="https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1547">https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1547</a>

This change introduces a behavior where limits where considered for each
variable independently. While not ideal, as lower resolutions might
support higher frame rates, and the way the build config ties
semantically the framerate to the resolution specified, an alternative
fix would require either an absolute max rate to be specified, or
possibly different sets of resolution + framerate limits, which might be
overkill to this purpose.

Original autor: Filipe Norte &lt;filipe_norte@comcast.com&gt;

* Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp:
(WebCore::GStreamerRegistryScanner::isContentTypeSupported const): Limit framerate even when the dimensions are smaller than the max limit set for them.

Canonical link: <a href="https://commits.webkit.org/297974@main">https://commits.webkit.org/297974@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6b3e444c155f89007120a21cbe1de35857173ce7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/113554 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33261 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23700 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119725 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/64316 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/115443 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33868 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41831 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86382 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/41457 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/116502 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27046 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102067 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66719 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26304 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20191 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63451 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96428 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20272 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/122960 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40557 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30300 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95235 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40948 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98277 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94988 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24255 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40111 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17922 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/36799 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40440 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/45942 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40099 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43411 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41856 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

philn reviewed (approved):
Would it be possible to eventually have this stuff covered by layout tests?

eocanha committed:
[GStreamer] Fix content type frame rate limit
https://bugs.webkit.org/show_bug.cgi?id=296572

Reviewed by Philippe Normand.

Decoder limits are set by a build configuration in the form of
widthxheight@framerate. However, framerate limit is only considered when
the width and height are the ones in the build config. If the actual
requested resolution is different, no framerate limit is ever
considered.

See: https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1547

This change introduces a behavior where limits where considered for each
variable independently. While not ideal, as lower resolutions might
support higher frame rates, and the way the build config ties
semantically the framerate to the resolution specified, an alternative
fix would require either an absolute max rate to be specified, or
possibly different sets of resolution + framerate limits, which might be
overkill to this purpose.

Original autor: Filipe Norte <filipe_norte@comcast.com>

* Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp:
(WebCore::GStreamerRegistryScanner::isContentTypeSupported const): Limit framerate even when the dimensions are smaller than the max limit set for them.

Canonical link: https://commits.webkit.org/297974@main
